openbas-api/src/main/java/io/openbas/execution/ExecutionExecutorService.java

```diff
@@ -27,30 +29,87 @@ public class ExecutionExecutorService {
   public void launchExecutorContext(Inject inject) {
     // First, get the agents of this injects
     List<Agent> agents = this.injectService.getAgentsByInject(inject);
+    // Filter each list to do something for each specific case and then remove the specific agents
+    // from the main "agents" list to execute payloads at the end for the remaining "normal" agents
+    List<Agent> inactiveAgents = agents.stream().filter(agent -> !agent.isActive()).toList();
+    agents.removeAll(inactiveAgents);
+    List<Agent> withoutExecutorAgent =
+        agents.stream().filter(agent -> agent.getExecutor() == null).toList();
+    agents.removeAll(withoutExecutorAgent);
+    List<Agent> crowdstrikeAgents =
+        agents.stream()
+            .filter(agent -> CROWDSTRIKE_EXECUTOR_TYPE.equals(agent.getExecutor().getType()))
+            .collect(Collectors.toList());
+    agents.removeAll(crowdstrikeAgents);
 
     InjectStatus injectStatus =
         inject.getStatus().orElseThrow(() -> new IllegalArgumentException("Status should exist"));
     AtomicBoolean atLeastOneExecution = new AtomicBoolean(false);
-    AtomicBoolean atOneTraceAdded = new AtomicBoolean(false);
-
+    AtomicBoolean atLeastOneTraceAdded = new AtomicBoolean(false);
+    // Manage inactive agents
+    if (!inactiveAgents.isEmpty()) {
+      inactiveAgents.forEach(
+          agent ->
+              injectStatus.addTrace(
+                  ExecutionTraceStatus.AGENT_INACTIVE,
+                  "Agent "
+                      + agent.getExecutedByUser()
+                      + " is inactive for the asset "
+                      + agent.getAsset().getName(),
+                  ExecutionTraceAction.COMPLETE,
+                  agent));
+      atLeastOneTraceAdded.set(true);
+    }
+    // Manage without executor agents
+    if (!withoutExecutorAgent.isEmpty()) {
+      inactiveAgents.forEach(
+          agent ->
+              injectStatus.addTrace(
+                  ExecutionTraceStatus.ERROR,
+                  "Cannot find the executor for the agent "
+                      + agent.getExecutedByUser()
+                      + " from the asset "
+                      + agent.getAsset().getName(),
+                  ExecutionTraceAction.COMPLETE,
+                  agent));
+      atLeastOneTraceAdded.set(true);
+    }
+    // Manage Crowdstrike agents for batch execution
+    try {
+      ExecutorContextService executorContextService =
+          context.getBean(CROWDSTRIKE_EXECUTOR_NAME, ExecutorContextService.class);
+      crowdstrikeAgents =
+          executorContextService.launchBatchExecutorSubprocess(
+              inject, crowdstrikeAgents, injectStatus);
+      atLeastOneExecution.set(true);
+    } catch (Exception e) {
+      crowdstrikeAgents.forEach(
```

Comment:
Should we log the exception in addition to inserting the traces, in order to have the full stack trace somewhere ?